client/src/pages/admin/tasks.tsx

```diff
@@ -108,6 +130,10 @@ function Page(props: Props) {
     [modalData, modalAction, modalLoading, dataCriteria],
   );
 
+  const allUsedCourses = uniqBy(data.map(({ courses }) => courses).flat(), course => course.name)
```

Comment:
can we use flatMap here?

Reply:
Good catch! Indeed `flatMap` is perfect here.
Fixed